mono/metadata/locales.c

```diff
@@ -530,22 +532,40 @@ get_current_locale_name (void)
 	return ret;
 }
 
-MonoString*
-ves_icall_System_Globalization_CultureInfo_get_current_locale_name (void)
+static void
+culture_info_get_current_locale_name (MonoStringHandle ret, MonoError *error)
 {
 	gchar *locale;
-	MonoString* ret;
 	MonoDomain *domain;
 
+	mono_error_init (error);
+
 	locale = get_current_locale_name ();
-	if (locale == NULL)
-		return NULL;
+	if (locale == NULL) {
+		MONO_HANDLE_ASSIGN (ret, NULL);
+		return;
+	}
 
 	domain = mono_domain_get ();
-	ret = mono_string_new (domain, locale);
+	MONO_PREPARE_GC_CRITICAL_REGION;
+	MONO_HANDLE_ASSIGN (ret, mono_string_new_checked (domain, locale, error));
+	MONO_FINISH_GC_CRITICAL_REGION;
 	g_free (locale);
+}
 
-	return ret;
+MonoString*
+ves_icall_System_Globalization_CultureInfo_get_current_locale_name (void)
+{
+	MonoError error;
+	MonoString *result_raw = NULL;
+	ICALL_ENTRY ();
+	MONO_HANDLE_ARENA_PUSH ();
+	MONO_HANDLE_DCL (MonoString, result);
+	culture_info_get_current_locale_name (result, &error);
+	MONO_HANDLE_ARENA_POP_RETURN_UNSAFE (result, result_raw);
+	ICALL_EXIT ();
+	mono_error_set_pending_exception (&error);
```

Comment:
we either should put that before `MONO_HANDLE_ARENA_POP_RETURN_UNSAFE (result, result_raw)`, or add a guarantee to the implementation that we do not hit a safepoint.